src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java

```diff
@@ -290,9 +385,103 @@ public WindowTownHall(final BuildingTownHall.View townHall)
         registerButton(BUTTON_CHANGE_SPEC, this::doNothing);
         registerButton(BUTTON_TOGGLE_JOB, this::toggleHiring);
 
+        registerButton(BUTTON_PREV_PAGE_PERM, this::switchPage);
+        registerButton(BUTTON_NEXT_PAGE_PERM, this::switchPage);
+
+        registerButton(BUTTON_MANAGE_OFFICER, this::editOfficer);
+        registerButton(BUTTON_MANAGE_FRIEND, this::editFriend);
+        registerButton(BUTTON_MANAGE_NEUTRAL, this::editNeutral);
+        registerButton(BUTTON_MANAGE_HOSTILE, this::editHostile);
+
+
         registerButton(BUTTON_UP, this::updatePriority);
         registerButton(BUTTON_DOWN, this::updatePriority);
         registerButton(BUTTON_DELETE, this::deleteWorkOrder);
+
+        findPaneOfTypeByID(BUTTON_PREV_PAGE_PERM, Button.class).setEnabled(false);
+        findPaneOfTypeByID(BUTTON_MANAGE_OFFICER, Button.class).setEnabled(false);
+        //todo set up al the buttons with messages for the permission changes!
+
+        registerButton(BUTTON_TRIGGER, this::trigger);
+
+    }
+
+    public void trigger(Button button)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Document this public method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUndocumentedApi)